### PR TITLE
[WIP] fix(android): empty frame stack after suspend/resume of application

### PR DESCRIPTION
**Pull request by vchimev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?
In the suspend/resume scenario, when the activity is destroyed and the application restores, navigation does not execute as the *JavaScript* instance of the current page is still alive. 
Therefore, the frame stack is `undefined`.

## What is the new behavior?
In this case, explicitly add the frame to the stack.

Fixes https://github.com/NativeScript/NativeScript/issues/7594.

### CODE REVIEW

File: e2e/nested-frame-navigation/tsconfig.json [hunk 1]
```diff
@@ -21,9 +21,14 @@
...
+    ],
     "exclude": [
+        "../../tns-core-modules/node_modules",
         "node_modules",
         "platforms",
         "e2e"
     ]
-}
\ No newline at end of file
+}
```
Comment (MartoYankov) [inline e2e/nested-frame-navigation/tsconfig.json]: Why is this needed?

File: tns-core-modules/ui/frame/frame.android.ts [hunk 2]
```diff
@@ -164,6 +164,7 @@ export class Frame extends FrameBase {
         // In this case call _navigateCore in order to recreate the current fragment.
         // Don't call navigate because it will fire navigation events.
         // As JS instances are alive it is already done for the current page.
+        // This requires explicitly to add the frame to the stack.
```
Comment (MartoYankov) [inline tns-core-modules/ui/frame/frame.android.ts]: I'm not sure I understand this comment. Can you elaborate?

File: e2e/nested-frame-navigation/e2e/issues.e2e-spec.ts [hunk 3]
```diff
@@ -30,27 +31,27 @@ describe("issues", async function () {
         }
     });
 
-    it("issue-6488", async function () {
-        await screen.loadedHome();
-        const showSomePage = async function () {
-            const somePageBtn = await driver.waitForElement("somePageOnRoot");
+    it("issues: 6488, 7594", async function () {
+        const navigateToSomePage = async function () {
+            const somePageBtn = await driver.waitForElement(somePage);
```
Comment (MartoYankov) [inline e2e/nested-frame-navigation/e2e/issues.e2e-spec.ts]: This isn't searching by the `automationText` value, but the `text` value.

Comment (vchimev) [reply e2e/nested-frame-navigation/e2e/issues.e2e-spec.ts]: Good eye!

File: e2e/nested-frame-navigation/tsconfig.json (hunk 1 above)
Comment (vchimev) [reply e2e/nested-frame-navigation/tsconfig.json]: It looks like it is not needed.

File: tns-core-modules/ui/frame/frame.android.ts (hunk 2 above)
Comment (vchimev) [reply tns-core-modules/ui/frame/frame.android.ts]: How do you understand it?

My understanding is that this block of comments explains the case at https://github.com/NativeScript/NativeScript/blob/6.0.0/tns-core-modules/ui/frame/frame.android.ts#L181,
when the application restores and navigation does not execute.

Therefore, I extended the block of comments to note that when navigation does not execute, the frame stack needs to be explicitly updated.

Comment (MartoYankov) [reply tns-core-modules/ui/frame/frame.android.ts]: Now I think maybe we shouldn't add this comment at all. Imho, L181 is self documenting and a comment might confuse another developer that there is something more to it that isn't obvious.

Comment (vchimev) [reply tns-core-modules/ui/frame/frame.android.ts]: Removed.

File: e2e/nested-frame-navigation/e2e/issues.e2e-spec.ts (hunk 3 above)
Comment (vchimev) [reply e2e/nested-frame-navigation/e2e/issues.e2e-spec.ts]: Updated application.

**Comment by vchimev** [comment]
There are test failures:
```
pr-integration-tests-nested-frame-navigation-frame-root-transition-None-scenarios-android23
pr-integration-tests-nested-frame-navigation-layout-root-transition-None-scenarios-android23
...
```

**Comment by vchimev** [comment]
The fix works for a single frame, however, it does not work for multi frames.

**Comment by vchimev** [comment]
I'm closing this pull request as the issue could be avoided following the [recommended approach](https://github.com/NativeScript/NativeScript/issues/7594#issuecomment-524818773) and due to higher priority/severity tasks.

**Comment by mannok** [comment]
Hi @vchimev , I am developing my app using multiple frame approach. What can I do to solve it?
